scanpipe/pipes/d2d.py

```diff
@@ -817,26 +822,119 @@ def _map_javascript_path_resource(
-            extra_data=extra_data,
+def map_javascript_colocation(project, logger=None):
+    """Map JavaScript files based on neighborhood file mapping."""
+    project_files = project.codebaseresources.files().only("path")
+
+    to_resources_key = (
+        project_files.to_codebase()
+        .no_status()
+        .filter(extension__in=[".map", ".ts"])
+        .exclude(name__startswith=".")
+        .exclude(path__contains="/node_modules/")
+    )
+
+    to_resources = project_files.to_codebase().no_status().exclude(name__startswith=".")
+
+    from_resources = project_files.from_codebase().exclude(path__contains="/test/")
+    resource_count = to_resources_key.count()
+
+    if logger:
+        logger(
+            f"Mapping {resource_count:,d} to/ resources against from/ codebase"
+            " based on neighborhood file mapping."
         )
-    return len(transpiled)
+
+    resource_iterator = to_resources_key.iterator(chunk_size=2000)
+    last_percent = 0
+    map_count = 0
+    start_time = timer()
+
+    for resource_index, to_resource in enumerate(resource_iterator):
+        last_percent = pipes.log_progress(
+            logger,
+            resource_index,
+            resource_count,
+            last_percent,
+            increment_percent=10,
+            start_time=start_time,
+        )
+        map_count += _map_javascript_colocation_resource(
+            to_resource, to_resources, from_resources, project
+        )
+
+    logger(f"{map_count:,d} resource(s) mapped")
+
+
+def _map_javascript_colocation_resource(
+    to_resource, to_resources, from_resources, project
+):
+    """Map JavaScript files based on neighborhood file mapping."""
+    path = to_resource.path
+
+    if to_resource.status or "-extract/" not in path:
+        return 0
+
+    coloaction_path, _ = path.rsplit("-extract/", 1)
+
+    neighboring_relations = project.codebaserelations.filter(
+        to_resource__path__startswith=coloaction_path,
+        map_type__in=["java_to_class", "js_compiled"],
+    )
+
+    if not neighboring_relations:
+        return 0
+
+    common_parent = neighboring_relations[0].from_resource.path
+    for relation in neighboring_relations:
+        s2 = relation.from_resource.path
+        common_parent, _ = common_prefix(common_parent, s2)
+
+    # No colocation mapping if the common parent is the root directory.
+    if not common_parent or len(Path(common_parent).parts) < 2:
+        return 0
+
+    from_neighboring_resources = from_resources.filter(path__startswith=common_parent)
+
+    try:
+        if sources := js.get_map_sources(to_resource):
+            from_resource = from_neighboring_resources.get(path__endswith=sources[0])
+            return js.map_related_files(
+                to_resources,
+                to_resource,
+                from_resource,
+                "js_colocation",
+                {},
+            )
+    except (MultipleObjectsReturned, ObjectDoesNotExist):
```

Comment:
You can use the `suppress` instead of except/pass.

Reply:
Also, there's too much happening here, we don't know which line may raise the exception.